app/src/main/java/org/mozilla/tv/firefox/webrender/cursor/CursorView.kt

```diff
@@ -5,89 +5,96 @@
 
-    fun updateCursorPressedState(event: KeyEvent) {
-        // Enter for keyboard and emulator use.
-        val remoteKey = RemoteKey.fromKeyEvent(event)
-        if (remoteKey == RemoteKey.CENTER || event.keyCode == KeyEvent.KEYCODE_ENTER) {
-            if (event.action == KeyEvent.ACTION_DOWN) {
-                setImageResource(R.drawable.cursor_full_active)
-            } else if (event.action == KeyEvent.ACTION_UP) {
-                setImageResource(R.drawable.cursor_full)
+    @CheckResult(suggest = "Dispose me, please. 🥰")
+    fun setup(cursorModel: CursorModel): Disposable {
+        this.cursorModel = cursorModel
+
+        val compositeDisposable = CompositeDisposable()
+
+        cursorModel.isCursorActive.subscribe { isCursorActive ->
+            isVisible = isCursorActive
+            if (!isCursorActive) {
+                animate().cancel()
             }
+        }.addTo(compositeDisposable)
+
+        Observables.combineLatest(cursorModel.isCursorMoving, cursorModel.isSelectPressed) {
+            // Only emit false if we are both stationary and not pressed
+            moving, pressed -> moving || pressed
         }
-    }
+                .distinctUntilChanged()
+                .subscribe { cursorIsMovingOrPressed ->
+                    if (cursorIsMovingOrPressed) {
+                        animate().cancel()
+                        alpha = 1f
+                        invalidate()
+                    } else {
+                        animate()
+                                .setStartDelay(HIDE_AFTER_MILLIS)
+                                .setDuration(HIDE_ANIMATION_DURATION_MILLIS)
+                                .alpha(0f)
+                                .start()
+                    }
+                }.addTo(compositeDisposable)
 
-    fun cancelUpdates() {
-        animate().cancel()
-        hideHandler.removeMessages(HIDE_MESSAGE_ID)
-    }
+        cursorModel.isSelectPressed
+                .distinctUntilChanged()
+                .subscribe { pressed -> when (pressed) {
+                        true -> setImageResource(BITMAP_PRESSED)
+                        false -> setImageResource(BITMAP_UNPRESSED)
+                    } }.addTo(compositeDisposable)
 
-    fun startUpdates() {
-        setMaxVisibility()
-        resetCountdown()
+        return compositeDisposable
     }
 
-    private fun setMaxVisibility() {
-        animate().cancel()
-        alpha = 1f
-    }
+    override fun onDraw(canvas: Canvas?) {
+        super.onDraw(canvas)
+        // The amount that this view must be offset for it to appear centered
+        // (otherwise the PointF we set would be its top left corner)
+        val xOffset = width / 2
+        val yOffset = height / 2
 
-    private fun resetCountdown() {
-        hideHandler.removeMessages(HIDE_MESSAGE_ID)
-        hideHandler.sendEmptyMessageDelayed(HIDE_MESSAGE_ID, HIDE_AFTER_MILLIS)
-    }
-}
+        onDrawMutablePositionCache.set(x + xOffset, y + yOffset)
```

Comment:
nit: rather than duplicating our offset logic (to add it and remove it), maybe we should just pass width & height to the `mutatePosition` method and let it handle the complexities:
```kotlin
fun mutatePosition(topLeftPosition: PointF, width: Int, height: Int) { ...

Reply:
It winds up being considerably more complex if we take that approach, because the model and the view have different conceptions of where the view actually is.  Touch events need to be at the center of the view, so this kind of one off calculation is always going to live somewhere.  It might be simpler to do when creating the touch event (at least we only calculate it once, instead of twice), but it's a marginal difference.

Reply:
> (at least we only calculate it once, instead of twice), but it's a marginal difference.

I'm concerned about this duplication (rather than the existence of this calculation): if the code changes and we change the calculation in one place, will we remember to update in the other? Without knowing what it'd look like, that biases me towards doing it in one place, but maybe it's too complex.

Reply:
Personally, I'm inclined to leave it as it is for that very reason.  The model considers the point to be 0,0, and the view offsets the display of that point.  I find that to be a reasonable division of concerns.